.ci/Jenkinsfile

```diff
@@ -17,7 +17,7 @@ pipeline {
     NOTIFY_TO = 'build-apm+apm-agent-nodejs@elastic.co'
     NPMRC_SECRET = 'secret/jenkins-ci/npmjs/elasticmachine'
     TOTP_SECRET = 'totp/code/npmjs-elasticmachine'
-    NODE_VERSION = 'v14.17.5'
+    BUILD_NODE_VERSION = 'v16.15.0'
```

Comment:
REVIEW NOTE: 
1. Switch to v16 for npm v8 which has `lockfileVersion: 2`.
2. Rename to `BUILD_NODE_VERSION` to differentiate from other unrelated usages of `NODE_VERSION` in this and other files in the repo.